File: 2021/src/test/java/nel/marco/Day10Test.java
package nel.marco;

import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.ArrayList;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

class Day10Test {

    Day10 day10 = new Day10();

    public List<String> readInput(boolean readExample) throws IOException {

        String path = "src\\main\\resources\\day10";
        if (readExample) {
            path = "src\\main\\resources\\day10Example";
        }

        List<String> stringList = Files.readAllLines(Path.of(path));

        return new ArrayList<>(stringList);

    }

    @Test
    void part1() throws IOException {

        assertEquals(26397, day10.part1(readInput(true)));
        assertEquals(339477, day10.part1(readInput(false)));
    }

    @Test
    void part2() throws IOException {
        assertEquals(288957, day10.part2(readInput(true)));
        assertEquals(3049320156L, day10.part2(readInput(false)));
    }
}

File: 2024/src/test/nel/marco/Day10Test.kt
package nel.marco

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNotEquals
import org.junit.jupiter.api.Test

class Day10Test {

    val day = Day10(useExample = false, useMac = true)
    val example = Day10(useExample = true, useMac = true)

    @Test
    fun answerOne() {
        assertEquals("X", example.answerOne())
        assertEquals("X", day.answerOne()) // not 6273684533700
    }

    @Test
    fun answerTwo() {
        assertEquals("X", example.answerTwo())
        assertEquals("X", day.answerTwo()) // cheated this one :(
    }

    @Test
    fun avg() {
        val day = day
        MarcoUtil.avgTime("part1", 5) {
            assertEquals("X", day.answerOne())
        }
        MarcoUtil.avgTime("part2", 5) {
            assertEquals("X", day.answerTwo())
        }
    }
}

